### PR TITLE
Moving ydk registry to cli engine

### PR DESCRIPTION
**Pull request by hotlib**
Moving ydk registry to cli engine

### CODE REVIEW

File: devmand/gateway/src/devmand/test/cli/StructuredUbntDeviceTest.cpp [hunk 1]
```diff
@@ -420,6 +420,8 @@ static const string EXPECTED_OUTPUT =
 
 TEST_F(StructuredUbntDeviceTest, DISABLED_getState) {
   devmand::Application app;
+  app.init();
```
Comment (marosmars) [inline devmand/gateway/src/devmand/test/cli/StructuredUbntDeviceTest.cpp]: We cannot rely on app.init here. 
Instead, inject the mregistry into the device via constructor (additional constructor overload might be needed)

**Review by marosmars — CHANGES_REQUESTED** [review]

File: devmand/gateway/src/devmand/test/cli/StructuredUbntDeviceTest.cpp (hunk 1 above)
Comment (hotlib) [reply devmand/gateway/src/devmand/test/cli/StructuredUbntDeviceTest.cpp]: We do not rely on app.init, we rely that the CliEngine will have mregistry (which must hold true as it does for other singleton objects like executors). app.init is just a means to get CliEngine initialized.

Maybe we could inject engine directly as has been done for some other tests.

**Review by tomasol — APPROVED** [review]

File: devmand/gateway/src/devmand/channels/cli/engine/Engine.h [hunk 2]
```diff
@@ -46,7 +48,7 @@ class Engine : public channels::Engine {
   shared_ptr<folly::Executor> kaCliExecutor;
 
   shared_ptr<folly::ThreadWheelTimekeeper> getTimekeeper();
-
+  shared_ptr<ModelRegistry> mreg;
```
Comment (marosmars) [inline devmand/gateway/src/devmand/channels/cli/engine/Engine.h]: should be private

**Review by marosmars — CHANGES_REQUESTED** [review]
1 minor change request

**Pull request closed without merging**